fix: add admin schema to control_plane config

jiangfucheng opened the pull request:
### Description

<!-- Please include a summary of the change and which issue is fixed. -->
<!-- Please also include relevant motivation and context. -->

Fixes https://github.com/apache/apisix/issues/8804

Maybe just add the schema of admin to conrol_plane config definition is enough?

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

<!--

Note

1. Mark the PR as draft until it's ready to be reviewed.
2. Always add/update tests for any changes unless you have a good reason.
3. Always update the documentation to reflect the changes made in the PR.
4. Make a new commit to resolve conversations instead of `push -f`.
5. To resolve merge conflicts, merge master instead of rebasing.
6. Use "request review" to notify the reviewer after making changes.
7. Only a reviewer can mark a conversation as resolved.

-->

spacewander reviewed (commented):
Can you add a test to validate the admin schema in https://github.com/apache/apisix/blob/master/t/cli/test_deployment_control_plane.sh?
So we won't miss schema next time.

jiangfucheng commented:
> Can you add a test to validate the admin schema in https://github.com/apache/apisix/blob/master/t/cli/test_deployment_control_plane.sh? So we won't miss schema next time.

I added test case for this pr. I wanted to through `schema.validate()` method to validate whether config is right, but admin field already in default-config.yaml, `schema.validate()` can't return any error, so i deleted admin field in config-default.yaml.

spacewander reviewed (commented):
You can provide an incorrect type in
https://github.com/apache/apisix/blob/e021266129eff537ea6b59cca0061385663d3a69/apisix/cli/schema.lua#L338

The basic rule: try your best to avoid touching the functional code when you write test

jiangfucheng commented:
@spacewander I can't reproduce the failed CI, could you guide me how to fix it?

shreemaan-abhishek commented:
@jiangfucheng You can make an empty commit and push it to re-run the CI. This might be some error that does not occur consistently.

spacewander commented on t/cli/test_deployment_control_plane.sh:
```diff
@@ -40,6 +40,37 @@ fi
+' > conf/config.yaml
+
+out=$(make init 2>&1 || true)
+if ! echo "$out" | grep 'expect: number, but got: string'; then
```
The err is from there: https://github.com/apache/apisix/blob/0bc65ea9acd726f79f80ae0abd8f50b7eb172e3d/apisix/cli/file.lua#L180
It is detected before the schema check.
We should find another way to validate the schema check.

jiangfucheng replied:
I changed a special field, it can be merged, but can't pass config validation, although this way is not perfect。🤣

spacewander commented on apisix/cli/schema.lua:
```diff
@@ -409,7 +410,7 @@ local deployment_schema = {
             },
         },
-        required = {"etcd", "role_control_plane"}
+        required = {"etcd", "admin", "role_control_plane"}
```
The admin is not required as we always have a default value.

spacewander commented on t/cli/test_deployment_control_plane.sh:
```diff
@@ -40,6 +40,36 @@ fi
+            cert: t/certs/mtls_server.crt
+            cert_key: t/certs/mtls_server.key
+    admin:
+      https_admin: "abc"
```
```suggestion
        https_admin: "abc"
```

jiangfucheng replied:
Fixed
